server/cmwell-dc/src/main/scala/cmwell/dc/stream/InfotonRetriever.scala

```diff
@@ -131,7 +131,8 @@ object InfotonRetriever extends LazyLogging {
                       throw WrongPathGotException(s"Got path ${path} from _out that was not in the uuids request bulk: ${
                         state._1.map(i => i.uuid.utf8String + ":" + i.base.path).mkString(",")}")
                     case Some((builder, curExtraData)) => {
-                      builder ++= (nquad ++ endln)
+                      val finalNquad = if(nquad.utf8String.isEmpty) nquad else nquad ++ endln
```

Comment:
1. Why not simply:
`if(nquad.utf8String.nonEmpty) builder ++= finalNquad`
It's an imperative part of the code - use normal if for it. Why adding an empty nquad?
1. you are practically converting every nquad to a string. But all you want to check is whether it's empty or not.